lib/utils/schematic/elkAutoLayout.ts

```diff
@@ -0,0 +1,161 @@
+import type { SoupUtilObjects } from "@tscircuit/soup-util"
+import Elk, { type ElkNode } from "elkjs"
+import { transformSchematicElement } from "@tscircuit/soup-util"
+import { translate } from "transformation-matrix"
+
+interface LayoutNode {
+  id: string
+  x: number
+  y: number
+  width: number
+  height: number
+  ports: Array<{
+    id: string
+    x: number
+    y: number
+  }>
+}
+
+export async function layoutSchematicWithElk(db: SoupUtilObjects) {
+  const components = db.schematic_component.list()
+  const sourceTraces = db.source_trace.list()
+  const ports = db.schematic_port.list()
+
+  const elk = new Elk({
+    workerUrl: "./node_modules/elkjs/lib/elk-worker.min.js",
```

Comment:
we'll need to load this via blobUrl probably (unfortunately)

will require bundling step, similar to @tscircuit/eval-webworker

I'm thinking maybe we create a separate package for this just to simplify where this code lives and to make as many test cases as possible, but not sure